packages/openapi-generator/src/parser/schema.ts

```diff
@@ -169,9 +169,17 @@ function parseEnumSchema(
   const type = schema.type ? getType(schema.type) : 'string';
   return {
     type,
-    enum: (schema.enum || []).map(entry =>
-      type === 'string' ? getEnumStringValue(entry) : entry
-    )
+    enum: (schema.enum || []).map(entry => {
+      if (type === 'string' && entry !== null) {
+        return getEnumStringValue(String(entry));
```

Comment:
A general question. In general we use `undefined` over `null` because the optional set it to undefined. Should we perhaps also use `undefined` instead of `null`. However `null` is closer to the enum definiotn. @marikaner  @jjtang1985 what do you think?

Reply:
In general, if the schema defines `null` as a valid enum value, I think we should just use it without any changes. The spec definition should be treated over any programming language (if possible) to avoid confusing.

In this case, `nullable` only allows `null` as a valid enum value.
So the type should look like:
```ts
param: 'v1' | 'v2' | null
```

On the other hand, there is another keyword called `required`. When setting to false, it's optional, so the type should look like below, where `undefined` is assigned if no value is provided:
```ts
param?: 'v1' | 'v2'
```

Reply:
That is a good point so we do:
- nullable = true add the null value to enum list. (as it is)
-  required = false make the enum parameter optional

In the schema.ts the `required`property is considered, but I am not sure if there is a test. So would be good to add a test in the schema.spec.ts to see if serialization and `required:false` leads to `?` schema file.

Reply:
Also fits our [styleguide](https://github.com/SAP/cloud-sdk-js/blob/main/STYLEGUIDE.md#null-and-undefined).